### PR TITLE
Add ad_group_membership resource

### PR DESCRIPTION
**Pull request by koikonom**
This resource allows us to assign AD objects such as users and groups
to a given group.

In this PR we are  also adding forceArray to `RunWinRMCommand` because
we need to work around PowerShell's `ConvertTo-Json` command
peculiarity. Specifically if there is only one input object it will
return a hash but if there are many it will return a list. The problem
is that sometimes we expect a list with a single item in it, in which
case we need to wrap the JSON document in `[]` so it looks like a list.

### CODE REVIEW

**Comment by koikonom** [comment]
Additional notes. 

Initially I attempted to support multiple ways of specifying members (SAM, GUID, SID, DN) but it was becoming too complicated on the Read() side making it unsafe to ship.

Instead users can use group data sources to specify groups that are not managed by Terraform.

Closes #33

File: ad/internal/winrmhelper/winrm_group_membership.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		return fmt.Errorf("while running %s: %s", operation, err)
+	} else if result.ExitCode != 0 {
+		return fmt.Errorf("command %s exited with a non-zero exit code(%d), stderr: %s, stdout: %s", operation, result.ExitCode, result.StdErr, result.Stdout)
+	}
+
+	return nil
+}
+
+func (g *GroupMembership) addGroupMembers(client *winrm.Client, members []*GroupMember) error {
+	return g.bulkGroupMembersOp(client, "Add-ADGroupMember", members)
```
Comment (manicminer) [inline ad/internal/winrmhelper/winrm_group_membership.go:106]: Is a zero-length check for `members` needed here?

**Review by manicminer — APPROVED** [review]
@koikonom I'm wondering if this might make sense as part of the `ad_group` resource, since the member list is declarative? However it LGTM! :)

File: ad/internal/winrmhelper/winrm_group_membership.go (hunk 1 above)
Comment (koikonom) [reply ad/internal/winrmhelper/winrm_group_membership.go:106]: You're right, I'll add this before I merge!

**Comment by koikonom** [comment]
I also wondered if this should be part of ad_group, but then I thought that making it its own resource would allow users to manage membership for groups no t managed by this provider.

**Pull request merged**